tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala

```diff
@@ -144,17 +144,21 @@ class Z3SolverContext(val config: SolverConfig) extends SolverContext with LazyL
   private val statisticsLock: ReentrantLock = new ReentrantLock()
   // start a new thread to collect statistics
   private val statisticsThread = new Thread(() => {
-    while (state == Running()) {
+    var interrupted = false
```

Comment:
There is [`Thread#isInterrupted`](https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#isInterrupted--), I don't think you need an extra local flag